pkg/providers/instancetype/instancetype.go

```diff
@@ -275,5 +274,16 @@ func (p *Provider) GetInstanceTypes(ctx context.Context) ([]*ec2.InstanceTypeInf
 	}
 	atomic.AddUint64(&p.instanceTypesSeqNum, 1)
 	p.cache.SetDefault(InstanceTypesCacheKey, instanceTypes)
+	p.cleanOldInstanceTypesFromCache()
```

Comment:
A few questions here:
1. Should the cache that stores the instance types just be its own cache, separate from the global data values that are being cached?
2. If we have a separate cache here, I think we can just leverage cache.Reset() to reset this data when we want to clear out old data rather than having to call delete across all of the keys
3. How does this affect calls where `GetInstanceTypes()` may be called multiple times within a loop (like a `Create()` loop). Re-getting the instance types might result in inconsistent data, though I think that this is the case today. We should probably improve this story though.